corehq/apps/locations/adjacencylist.py

```diff
@@ -64,7 +64,7 @@ def make_cte_query(cte):
                 _depth=Value(0, output_field=field),
             ).union(
                 cte.join(
-                    self.all().order_by(),
+                    self.filter(domain=node.domain).order_by(),
```

Comment:
From an interface point of view this is not the best way, since in theory, `AdjListModel` could be used to represent any other tree objects that may not have domain attribute, but currently we only use it for locations. And also most of our data models are expected to have a domain attribute. So, I thought this is not such a bad idea. Open to do tweak this, if there are any substantially better ideas.